### PR TITLE
CMS-42 Provide translation support for Stories, Features, Statistics, Headers, Roadmap Iterations

### PR DESCRIPTION
**Pull request by ex1st0r**

### CODE REVIEW

File: pages/faq.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ import styles from './faq.sass'
 
 class FAQ extends React.Component {
 
-  static async getInitialProps ({ store }) {
+  static async getInitialProps ({ store, req }) {
+    await store.dispatch(setUserLanguages(req && req.headers && req.headers['accept-language']))
```
Comment (ipavlenko) [inline pages/faq.js]: Send req.headers to setUserLanguages, it should also check Cookies in the future.

File: src/models/lib/HeaderModel.js [hunk 2]
```diff
@@ -48,15 +49,17 @@ export default class HeaderModel {
...
+
     return data == null ? data : new HeaderModel({
       id: data._id,
       slug: data.slug,
       title: data.title,
       stereotype: data.stereotype,
       background: data.background,
       video: data.video,
-      brief: data.brief,
+      brief: localeModelFields && 'brief' in localeModelFields ? localeModelFields.brief : data.brief ,
```
Comment (ipavlenko) [inline src/models/lib/HeaderModel.js:62]: Did you forget (-brackets-) for 'brief' in localeModelFields ?

File: src/models/lib/IterationModel.js [hunk 3]
```diff
@@ -21,11 +22,13 @@ export default class IterationModel {
     })
   }
 
-  static fromServerModel (data) {
+  static fromServerModel (data, { locales }) {
+    let localeModelFields = getLocaleModelFields(data, locales)
+
     return data == null ? data : new IterationModel({
       id: data._id,
       title: data.title,
```
Comment (ipavlenko) [inline src/models/lib/IterationModel.js]: Localize title

File: src/models/lib/StatisticModel.js [hunk 4]
```diff
@@ -18,11 +19,13 @@ export default class StatisticModel {
     })
   }
 
-  static fromServerModel (data) {
+  static fromServerModel (data, { locales }) {
+    let localeModelFields = getLocaleModelFields(data, locales)
+
     return data == null ? data : new StatisticModel({
       id: data._id,
       title: data.title,
```
Comment (ipavlenko) [inline src/models/lib/StatisticModel.js]: Localize title

File: src/models/lib/StoryModel.js [hunk 5]
```diff
@@ -24,16 +25,18 @@ export default class StoryModel {
     })
   }
 
-  static fromServerModel (data) {
+  static fromServerModel (data, { locales }) {
+    let localeModelFields = getLocaleModelFields(data, locales)
+
     return data == null ? null : new StoryModel({
       id: data._id,
       title: data.title,
```
Comment (ipavlenko) [inline src/models/lib/StoryModel.js]: Localize title

File: src/store/lib/pages/actions.js [hunk 6]
```diff
@@ -54,13 +56,16 @@ export const initMenus = () => async (dispatch, getState) => {
...
+  const locales = state.pages.userLocales
+
   if (state.pages.features.isLoaded) {
     return
   }
   const { data } = await BACKEND.get('features')
   return dispatch({
     type: PAGES_INIT_FEATURES,
-    features: data.features.map(FeatureModel.fromServerModel)
+    features: data.features.map((feature) => FeatureModel.fromServerModel(feature, { locales }))
```
Comment (ipavlenko) [inline src/store/lib/pages/actions.js:68]: Maybe it is better to find best locale here and pass only the best locale down to the fromServerModel method?

File: src/store/lib/pages/actions.js [hunk 7]
```diff
@@ -90,13 +95,16 @@ export const initJobs = () => async (dispatch, getState) => {
...
+  const locales = state.pages.userLocales
+
   if (state.pages.statistics.isLoaded) {
     return
   }
   const { data } = await BACKEND.get('statistics')
   return dispatch({
     type: PAGES_INIT_STATISTICS,
-    statistics: data.statistics.map(StatisticModel.fromServerModel)
+    statistics: data.statistics.map((statistic) => StatisticModel.fromServerModel(statistic, { locales }))
```
Comment (ipavlenko) [inline src/store/lib/pages/actions.js:107]: See comments above

File: src/store/lib/pages/actions.js [hunk 8]
```diff
@@ -114,13 +122,16 @@ export const initMembers = () => async (dispatch, getState) => {
...
+  const locales = state.pages.userLocales
+
   if (state.pages.headers.isLoaded) {
     return
   }
   const { data } = await BACKEND.get('headers')
   return dispatch({
     type: PAGES_INIT_HEADERS,
-    headers: data.headers.map(HeaderModel.fromServerModel)
+    headers: data.headers.map((header) => HeaderModel.fromServerModel(header, { locales }))
```
Comment (ipavlenko) [inline src/store/lib/pages/actions.js:134]: See comments above

File: src/store/lib/pages/actions.js [hunk 9]
```diff
@@ -150,13 +161,16 @@ export const initProducts = () => async (dispatch, getState) => {
...
+  const locales = state.pages.userLocales
+
   if (state.pages.stories.isLoaded) {
     return
   }
   const { data } = await BACKEND.get('stories')
   return dispatch({
     type: PAGES_INIT_STORIES,
-    stories: data.stories.map(StoryModel.fromServerModel)
+    stories: data.stories.map((story) => StoryModel.fromServerModel(story, { locales }))
```
Comment (ipavlenko) [inline src/store/lib/pages/actions.js:173]: See comments above

File: src/store/lib/pages/actions.js [hunk 10]
```diff
@@ -186,13 +200,16 @@ export const initTestimonials = () => async (dispatch, getState) => {
...
+  const locales = state.pages.userLocales
+
   if (state.pages.iterations.isLoaded) {
     return
   }
   const { data } = await BACKEND.get('iterations')
   return dispatch({
     type: PAGES_INIT_ITERATIONS,
-    iterations: data.iterations.map(IterationModel.fromServerModel)
+    iterations: data.iterations.map((iteration) => IterationModel.fromServerModel(iteration, { locales }))
```
Comment (ipavlenko) [inline src/store/lib/pages/actions.js:212]: See comments above

File: src/store/lib/pages/reducer.js [hunk 11]
```diff
@@ -45,6 +45,7 @@ const initialState = {
   statistics: makeArrayState(false, []),
   stories: makeArrayState(false, []),
   testimonials: makeArrayState(false, []),
+  userLocales: ''
```
Comment (ipavlenko) [inline src/store/lib/pages/reducer.js:48]: Maybe it is better to store best locale here?

**Review by ipavlenko — CHANGES_REQUESTED** [review]

File: src/store/lib/pages/actions.js (hunk 6 above)
Comment (ex1st0r) [reply src/store/lib/pages/actions.js:68]: We need list of locales for identify user language, but it individual for every model.

File: src/models/lib/HeaderModel.js (hunk 2 above)
Comment (ex1st0r) [reply src/models/lib/HeaderModel.js:62]: it's not necessary

File: src/store/lib/pages/reducer.js (hunk 11 above)
Comment (ex1st0r) [reply src/store/lib/pages/reducer.js:48]: see above

**Pull request merged**